src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt

```diff
@@ -65,26 +70,42 @@ class SignServiceImpl(
 
         val quotes = quoteService.getQuotes(quoteIds)
         quotes.forEach { quote ->
-            quote.memberId?.let { quoteMemberId ->
-                if (memberId != quoteMemberId) {
-                    logger.info("Member [id: $memberId] tried to sign quote with member id: $quoteMemberId. [Quotes: $quotes]")
-                    return StartSignErrors.variousMemberId
-                }
-            } ?: run {
-                logger.info("Member [id: $memberId] tried to sign quote without member id. [Quotes: $quotes]")
+            if (quote.memberId == null) {
+                logger.error("Member [id: $memberId] tried to sign quote without memberId [Quotes: $quotes]")
                 return StartSignErrors.noMemberIdOnQuote
             }
+            if (memberId != quote.memberId) {
+                logger.error("Member [id: $memberId] tried to sign quote with mismatching memberId [Quotes: $quotes]")
+                return StartSignErrors.variousMemberId
+            }
 
             val quoteNotSignableErrorDto = assertQuoteIsNotSignedOrExpired(quote)
             if (quoteNotSignableErrorDto != null) {
+                logger.error("Member [id: $memberId] tried to sign quote that was either signed or expired [Quotes: $quotes]")
                 return StartSignErrors.fromErrorResponse(quoteNotSignableErrorDto)
             }
         }
 
-        return signStrategyService.startSign(
+        val personalInfoMatching = quotes.windowed(2).all { (predecessor, successor) ->
```

Comment:
Not sure if the words "predecessor, successor" is the correct choice here, since we are just checking if the info is matching between quotes and the quotes are not succeeding one another. (My mother tongue is Greek, so i may be completely wrong). Plus we can extract it to a private function.
```suggestion
        val personalInfoMatching = quotes.windowed(2).all { (left, right) ->
```

Reply:
I think the naming is decent based on the function name "windowed" that passes through the list, thus one of the quotes will be the "predecessor" and the other the "successor" of each other. The "successor" will be the "predecessor" when it moves one more step. But "left" and "right" could be sufficient if you imagine it moving from left to right